sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultServiceSpec.java

```diff
@@ -75,10 +75,23 @@ public DefaultServiceSpec(
                 ? DcosConstants.MESOS_MASTER_ZK_CONNECTION_STRING : zookeeperConnection;
         this.pods = pods;
         this.replacementFailurePolicy = replacementFailurePolicy;
-        this.user = user;
+        this.user = getUser(user, pods);
         ValidationUtils.validate(this);
     }
 
+    @VisibleForTesting
+    static String getUser(String user, List<PodSpec> podSpecs) {
+        if (!StringUtils.isBlank(user)) {
+            return user;
+        }
+
+        if (podSpecs.get(0).getUser().isPresent()) {
```

Comment:
If `podSpecs` is an empty list, then this is going to throw on the call to `get(0)`.

We could also use:
```java
return podSpecs.get(0).getUser().orElse(DcosConstants.DEFAULT_SERVICE_USER);
```
instead of the `isPresent()` check. (this is more a question for me to get to know the preferred style).